### PR TITLE
Change error message when conflicting Vlan ID is used

### PR DESCRIPTION
**Pull request by sumukhatv**

### CODE REVIEW

File: go-server-server/go/default.go [hunk 1]
```diff
@@ -280,8 +280,13 @@ func ConfigInterfaceVlanPost(w http.ResponseWriter, r *http.Request) {
         return
     }
     if vlan_kv != nil {
-        WriteRequestErrorWithSubCode(w, http.StatusConflict, RESRC_EXISTS,
-              "Object already exists: " + vlan_name, []string{}, "")
+        vlan_if_kv, _ := GetKVs(db.db_num, generateDBTableKey(db.separator, VLAN_INTF_TB, vlan_name))
+        if vlan_if_kv != nil {
+                vnet_kv, _ := GetKVs(db.db_num, generateDBTableKey(db.separator, VNET_TB, vlan_if_kv["vnet_name"]))
```
Comment (prsunny) [inline go-server-server/go/default.go:285]: I thot we have a cache to find guid from vnet name.

File: go-server-server/go/default.go [hunk 2]
```diff
@@ -280,8 +280,13 @@ func ConfigInterfaceVlanPost(w http.ResponseWriter, r *http.Request) {
...
+                attr.Vnet_id = vnet_kv["guid"]
+                WriteRequestErrorWithSubCode(w, http.StatusConflict, RESRC_EXISTS,
+                    "Object already exists: {\"vlan_name\":\"" + vlan_name + "\", \"vnet_id\":\"" + vnet_kv["guid"] +"\"}", []string{}, "")
+        }
```
Comment (prsunny) [inline go-server-server/go/default.go:289]: please have the old error message in the 'else' part. Lets not return empty if vlan_if_kv is nil

File: go-server-server/go/default.go (hunk 1 above)
Comment (sumukhatv) [reply go-server-server/go/default.go:285]: We don't have that mapping in Cache.
All Caches we have are here: https://github.com/sonic-net/sonic-restapi/blob/043b33cd0f0e533aba45efd81f163fc1bb7f5991/go-server-server/go/persistent.go#L440

File: go-server-server/go/default.go [hunk 3]
```diff
@@ -280,8 +280,15 @@ func ConfigInterfaceVlanPost(w http.ResponseWriter, r *http.Request) {
         return
     }
     if vlan_kv != nil {
-        WriteRequestErrorWithSubCode(w, http.StatusConflict, RESRC_EXISTS,
-              "Object already exists: " + vlan_name, []string{}, "")
+        vlan_if_kv, _ := GetKVs(db.db_num, generateDBTableKey(db.separator, VLAN_INTF_TB, vlan_name))
+        if vlan_if_kv != nil {
+            vnet_kv, _ := GetKVs(db.db_num, generateDBTableKey(db.separator, VNET_TB, vlan_if_kv["vnet_name"]))
```
Comment (prsunny) [inline go-server-server/go/default.go:285]: can we check if vnet_name exists in the vlan_if_kv table. It is possible that some stale entry without vnet_name may exists .

File: go-server-server/go/default.go [hunk 4]
```diff
@@ -280,8 +280,18 @@ func ConfigInterfaceVlanPost(w http.ResponseWriter, r *http.Request) {
...
     if vlan_kv != nil {
+        vlan_if_kv, _ := GetKVs(db.db_num, generateDBTableKey(db.separator, VLAN_INTF_TB, vlan_name))
+        if vlan_if_kv != nil {
+            if vnet_name, ok := vlan_if_kv["vnet_name"]; ok {
+                vnet_kv, _ := GetKVs(db.db_num, generateDBTableKey(db.separator, VNET_TB, vnet_name))
+                WriteRequestErrorWithSubCode(w, http.StatusConflict, RESRC_EXISTS,
+                    "Object already exists: {\"vlan_name\":\"" + vlan_name + "\", \"vnet_id\":\"" + 
+                    vnet_kv["guid"] +"\"}", []string{}, "")
+                return
+            }
```
Comment (prsunny) [inline go-server-server/go/default.go:291]: this else also we should return error message but with empty vnet_id

Comment (sumukhatv) [reply go-server-server/go/default.go:291]: which else are you referring to? if `vlan_kv` is nil then the rest of code block needs to be executed. We cannot return an error message. if `vlan_if_kv` is nil, we return the previous error message. If `vnet_name` key is missing in `vlan_if_kv`, then also we return the previous error message.

Comment (prsunny) [reply go-server-server/go/default.go:291]: i'm talking about if vnet_name == ""

**Review by prsunny — APPROVED** [review]

**Pull request merged**